### ISSUE TITLE
[BUG] Safer app displays orange status for vaccinated users

### ISSUE THREAD

**Issue by mihail-varbanov** [rokwire/safer-illinois-app#761]
From @InnovativeInventor:

I can fully reproduce this bug. I am fully vaccinated and have tested recently, but the app showed orange for me on initial setup. Additionally, the [online boarding pass](https://icard.uillinois.edu/CCBoardingPass/index.cfm) shows my proper status (green). After setup, "Your COVID-19 Event History" screen is blank (it says "no history"). However, after manually clicking "Request my vaccine and latest test" button inside the history screen, the app behaves as expected (my status becomes green).

I think the issue is that the app is not properly fetching the testing/vaccine data on the setup flow (and it's unclear, from a user's perspective, how to trigger this fetch manually). I agree that this is absolutely a show-stopper. The whole point of the setup flow is to set up the app. I also don't think that https://github.com/rokwire/safer-illinois-app/commit/58020006ef7865383b5a7f4357c64faae764c224 helps this issue at all.

Please note that while I think this may be the same bug as @Priyaravi8 described, it may be a distinct, new bug.

I am on version `2.11.5` of the app, which I believe is the latest version released on the iOS app store.

(https://github.com/rokwire/safer-illinois-app/issues/755#issuecomment-962409335)

**Comment by mihail-varbanov** [comment]
From @InnovativeInventor,

Let me please first thank you for the clear and detailed report!

#755 addresses a problem in Safer Illinois 2.12 that uses different health rules (with new features added) than the version that you use - 2.11.5. My believe is that what you described is a different bug, not related to #755, and that's why I created a separate GIT issue for it.

Could you please answer:
1. Have you ever used the Safer Illinois application before on this or other device using the same account (NetId) as now?
2. Did the onboarding flow report a problem with your "COVID-19 Secret Key" (a RSA key pair) offering options to scan it from another device, to load it from saved pictures, or to reset it losing the testing/vaccination history?

CC @pmarkhennessy.

**Comment by InnovativeInventor** [comment]
> 1. Have you ever used the Safer Illinois application before on this or other device using the same account (NetId) as now?

Yes, but it didn't work for me. I believe there was something buggy with the app when I signed in with the provisional Illinois account that they give to prospective students that made it impossible to log in with my real NetID. Only after deleting and re-installing the app, I was able to log in with my real NetID.

> 2. Did the onboarding flow report a problem with your "COVID-19 Secret Key" (a RSA key pair) offering options to scan it from another device, to load it from saved pictures, or to reset it losing the testing/vaccination history?

No.

**Comment by mihail-varbanov** [comment]
Thanks @InnovativeInventor! My feeling is that problem occurred because of the accounts switching. For Health BB the "provisional Illinois account" and the "real NetID" should be two different users unless they have the same UIN. That explains why on first login with the "real NetID" the health history had been empty. So I would say the Safer Illinois application behaved properly.

CC @pmarkhennessy
